dev: suggest crosslinuxarm on arm64

RaduBerinde opened the pull request:
Update to `dev doctor` to suggest `crosslinuxarm` instead of
`crosslinux` if `GOARCH=arm64`.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/134344)
<!-- Reviewable:end -->

rickystewart commented on pkg/cmd/dev/doctor.go:
```diff
@@ -95,6 +96,30 @@ func (cfg *doctorConfig) maybePromptForAutofixPermission(question string) {
+		{name: "dev", description: "uses the host toolchain"},
+	}
+	if runtime.GOOS == "linux" {
+		crossCfg := "crosslinux"
```
(While it doesn't matter as the build is unlikely to work on these non-`amd64`, non-`arm64` systems, it's best if the code is written to be flexible rather than definitely broken.)

RaduBerinde replied:
TFTR! Good idea, done.

RaduBerinde committed:
dev: suggest crosslinuxarm on arm64

Update to `dev doctor` to suggest `crosslinuxarm` instead of
`crosslinux` if `GOARCH=arm64`.

Epic: none
Release note: None

RaduBerinde commented on pkg/cmd/dev/doctor.go:
```diff
@@ -95,6 +96,30 @@ func (cfg *doctorConfig) maybePromptForAutofixPermission(question string) {
+
+// buildConfigs returns the available build configs (typically "dev" and
+// "crosslinux"). The first config will always be "dev".
+func buildConfigs() []buildConfig {
```
Done.

RaduBerinde commented:
bors r+
